### PR TITLE
Extend the SQL UDFs and update repo state for v2

### PR DESCRIPTION
**Pull request by NicoK**
- fixed: a few special cases when matching components from file names
- new: SQL UDF `GetSourceComponent`
- new: SQL UDF `NormalizeEmailThread`
- new: SQL UDF `GetJiraTicketComponents`

---
This change is [Reviewable](https://reviewable.io/reviews/ververica/lab-flink-repository-analytics/13)

### CODE REVIEW

File: common/src/main/java/com/ververica/platform/Utils.java [hunk 1]
```diff
@@ -13,8 +13,12 @@
 import java.time.format.SignStyle;
 import java.time.temporal.ChronoField;
 import java.util.Date;
+import java.util.regex.Pattern;
 
 public class Utils {
+  public static Pattern SOURCE_FILENAME_COMPONENT_PATTERN =
```
Comment (sjwiesman) [inline common/src/main/java/com/ververica/platform/Utils.java:19]: Move this to `PatternUtils`?

**Review by sjwiesman — APPROVED** [review]
One small suggestion but otherwise this looks really good. Appreciate using regex named matches over numbers!

File: common/src/main/java/com/ververica/platform/Utils.java (hunk 1 above)
Comment (NicoK) [reply common/src/main/java/com/ververica/platform/Utils.java:19]: I wanted to but that's in a different subproject...I could move that to 'common' as well though

Comment (sjwiesman) [reply common/src/main/java/com/ververica/platform/Utils.java:19]: I didn’t realize it was a different sub project, feel free to leave it

Comment (NicoK) [reply common/src/main/java/com/ververica/platform/Utils.java]: I moved it around...going ahead with the merge

**Pull request merged**